src/op_mode/vpn.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import sys
+from vyos.util import call
+import vyos.opmode
+
+
+SWANCTL_CONF = '/etc/swanctl/swanctl.conf'
+
+
+def get_peer_connections(peer, tunnel, return_all = False):
+    peer = peer.replace(':', '-')
+    search = rf'^[\s]*(peer_{peer}_(tunnel_[\d]+|vti)).*'
+    matches = []
+    with open(SWANCTL_CONF, 'r') as f:
+        for line in f.readlines():
+            result = re.match(search, line)
+            if result:
+                suffix = f'tunnel_{tunnel}' if tunnel.isnumeric() else tunnel
+                if return_all or (result[2] == suffix):
+                    matches.append(result[1])
+    return matches
+
+
+def reset(protocol: str, peer: str, tunnel:str):
```

Comment:
I think the file should be `ipsec.py` and there shouldn't be a protocol option. Different VPN protocols will have entirely different operations and arguments, so trying to merge them together will likely end up confusing for the users and for developers alike. 

I also wonder if the operation should even be `reset_peer`.

There's also `reset vpn ipsec profile`, which is an entirely different operation. When we get to adding it, the `reset` function will need two _mutually-exclusive_ parameters in the current approach.

I believe we should make it `reset_peer` to make room for other operations right away.

Reply:
Done!